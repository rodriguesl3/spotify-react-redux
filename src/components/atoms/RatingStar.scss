.stars-inner {
    position: absolute;
    top: 0;
    left: 0;
    white-space: nowrap;
    overflow: hidden;
    width: 0;
}

.stars-outer {
    display: inline-block;
    position: relative;
    font-family: FontAwesome;
}

.stars-outer::before {
    content: "\2605 \2605 \2605 \2605 \2605";
}

.stars-inner::before {
    content: "\2605 \2605 \2605 \2605 \2605";
    color: #f8ce0b;
}