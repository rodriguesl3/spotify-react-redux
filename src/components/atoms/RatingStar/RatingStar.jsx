import React from 'react'

import './RatingStar.scss';

export const RatingStart = (props) => {
    const { ratingValue } = props;

    return (
        <div>
            <div class="stars-outer">
                <div class="stars-inner" style={{width: ratingValue+"%"}}></div>
            </div>
        </div>
    )
}