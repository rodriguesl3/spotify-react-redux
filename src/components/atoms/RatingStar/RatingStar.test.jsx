/* eslint-disable no-undef */
import React from 'react';
import renderer from 'react-test-renderer';
import { RatingStart } from './RatingStar';

test('RatingStart snaphot test', () => {
  const component = renderer.create(<RatingStart ratingValue="7.5" />);
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
