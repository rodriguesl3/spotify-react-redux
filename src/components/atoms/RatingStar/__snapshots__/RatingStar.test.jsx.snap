// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RatingStart snaphot test 1`] = `
<div>
  <div
    class="stars-outer"
  >
    <div
      class="stars-inner"
      style={
        Object {
          "width": "7.5%",
        }
      }
    />
  </div>
</div>
`;
